includes/admin/services/class-rop-facebook-service.php

```diff
@@ -577,36 +577,166 @@ private function prepare_for_sharing( $post_details ) {
+            // page was added using user application (old method)
+            $api = $this->get_api();
+            try {
+                $api->post( $path, $new_post, $token );
+
+                return true;
+            } catch ( Facebook\Exceptions\FacebookResponseException $e ) {
+                $errorMsg = $e->getMessage();
+
+                if ( strpos($errorMsg, '(#100)' ) !== false && !empty( $new_post['name'] )) {
+                    // https://developers.facebook.com/docs/graph-api/reference/v3.2/page/feed#custom-image
+                    // retry without name
+                    unset( $new_post['name'] );
+
+                    try {
+                        $api->post( $path, $new_post, $token );
+                        return true;
+                    } catch ( Facebook\Exceptions\FacebookResponseException $e ) {
+                        $this->logger->alert_error('Unable to share post for facebook. (FacebookResponseException) Error: ' . $e->getMessage());
+
+                        return false;
+                    } catch ( Facebook\Exceptions\FacebookSDKException $e ) {
+                        $this->logger->alert_error('Unable to share post for facebook.  Error: ' . $e->getMessage());
+
+                        return false;
+                    }
+
+                } else {
+                    $this->logger->alert_error( 'Unable to share post for facebook. (FacebookResponseException) Error: ' . $errorMsg);
+
+                    return false;
+                }
+            } catch ( Facebook\Exceptions\FacebookSDKException $e ) {
+                $this->logger->alert_error( 'Unable to share post for facebook.  Error: ' . $e->getMessage() );
+
+                return false;
+            }
+        } else {
+            // page was added using ROP application (new method)
+
+            $post_data = $new_post;
+            $post_data['access_token'] = $token;
+
+            if ( $posting_type === 'video' ) {
+                $url = 'https://graph-video.facebook.com' . $path;
+            } else {
+                $url = 'https://graph.facebook.com' . $path;
+            }
+
+            $ch = curl_init();
+            curl_setopt($ch, CURLOPT_URL,$url);
+            curl_setopt($ch, CURLOPT_POST, 1);
+            curl_setopt($ch, CURLOPT_HTTPHEADER, array('Content-Type: application/x-www-form-urlencoded'));
+            curl_setopt($ch, CURLOPT_POSTFIELDS, http_build_query($post_data));
+            curl_setopt($ch, CURLOPT_RETURNTRANSFER, true);
+
+            $response = curl_exec($ch);
+            $response_code = curl_getinfo($ch, CURLINFO_HTTP_CODE);
+
+            curl_close ($ch);
+
+            if ( $response_code === 200 ) {
+                return true;
+            } else if ( $response_code === 400 && strpos($response, '(#100)') !== false && !empty($new_post['name'])) {
+                // retry without name
+                unset($post_data['name']);
+                $ch = curl_init();
```

Comment:
To keep your changes more inline with the plugin's current setup can you try accomplishing the same thing here but using Guzzle instead? It's already included in the plugin